src/Simplex/Chat.hs

```diff
@@ -4489,6 +4517,21 @@ processAgentMessageConn user@User {userId} corrId agentConnId agentMessage = do
           groupMsgToView g' m ci msgMeta
         createGroupFeatureChangedItems user cd CIRcvGroupFeature g g'
 
+    xGrpDirectInv :: GroupInfo -> GroupMember -> ConnReqInvitation -> Maybe MsgContent -> RcvMessage -> MsgMeta -> m ()
+    xGrpDirectInv g m connReq mContent_ msg msgMeta = do
+      -- check direct messages preference
+      -- auto-join / client setting? (new to_join field if auto join is in UI?)
```

Comment:
yes, global setting with group overrides, on by default? We have some framework for them, they are in ChatSettings as separate fields.